shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLStatementSQLVisitor.java

```diff
@@ -605,7 +608,21 @@ public ASTNode visitQueryExpressionBody(final QueryExpressionBodyContext ctx) {
         if (1 == ctx.getChildCount() && ctx.getChild(0) instanceof QueryPrimaryContext) {
             return visit(ctx.queryPrimary());
         }
-        throw new IllegalStateException("union select is not supported yet.");
+        if (null != ctx.queryExpressionBody()) {
+            MySQLSelectStatement result = (MySQLSelectStatement) visit(ctx.queryExpressionBody());
+            result.getUnionSegments().add((UnionSegment) visitUnionClaus(ctx.unionClaus()));
+            return result;
+        }
+        MySQLSelectStatement result = (MySQLSelectStatement) visit(ctx.queryExpressionParens());
+        result.getUnionSegments().add((UnionSegment) visitUnionClaus(ctx.unionClaus()));
+        return result;
+    }
+    
+    @Override
+    public ASTNode visitUnionClaus(final UnionClausContext ctx) {
+        return new UnionSegment((null != ctx.unionOption() && null != ctx.unionOption().ALL()) ? UnionType.UNION_ALL : UnionType.UNION_DISTINCT,
```

Comment:
Please extract a variable to avoid the complexity of UnionSegment initialization.